arch/toolchains/eurohpc-meluxina-nvhpc.cmake

```diff
@@ -0,0 +1,59 @@
+# (C) Copyright 1988- ECMWF.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+####################################################################
+# COMPILER
+####################################################################
+
+set( ECBUILD_FIND_MPI ON )
+
+####################################################################
+# OpenMP FLAGS
+####################################################################
+
+# Note: OpenMP_Fortran_FLAGS gets overwritten by the FindOpenMP module
+# unless its stored as a cache variable
+set( OpenMP_Fortran_FLAGS   "-mp -mp=gpu,bind,allcores,numa" CACHE STRING "" )
+
+# Note: OpenMP_C_FLAGS and OpenMP_C_LIB_NAMES have to be provided _both_ to
+# keep FindOpenMP from overwriting the FLAGS variable (the cache entry alone
+# doesn't have any effect here as the module uses FORCE to overwrite the
+# existing value)
+set( OpenMP_C_FLAGS         "-mp -mp=bind,allcores,numa" CACHE STRING "" )
+set( OpenMP_C_LIB_NAMES     "acchost" CACHE STRING "")
+
+####################################################################
+# OpenAcc FLAGS
+####################################################################
+
+# NB: We have to add `-mp` again to avoid undefined symbols during linking
+# (smells like an Nvidia bug)
+set( OpenACC_Fortran_FLAGS "-acc=gpu -mp=gpu -gpu=cc80,lineinfo,fastmath" CACHE STRING "" )
```

Comment:
What undef symbols do you get here without the -mp?

Reply:
This is the bug I raised with Nvidia a few weeks ago:
```
nvlink error   : Undefined reference to '__hxdTargetTeamsEnd' in 'CMakeFiles/dwarf-cloudsc-gpu-omp-scc-hoist.dir/cloudsc_driver_gpu_omp_scc_hoist_mod.F90.o'
nvlink error   : Undefined reference to '__hxdTargetTeamsBegin' in 'CMakeFiles/dwarf-cloudsc-gpu-omp-scc-hoist.dir/cloudsc_driver_gpu_omp_scc_hoist_mod.F90.o'
```
It was rather sensitive to order of arguments when having both OpenMP offload and OpenACC enabled in the compiler flags. It isn't trivial to reproduce but the above worked reliably.
Supposedly it has been fixed with 22.7 but I haven't had a chance to confirm, yet. I have a standalone tarball with a configuration to reproduce the issue, which they used to test internally.

So hopefully this can go in a few months.